bin/build.sh

```diff
@@ -7,118 +7,92 @@ cd "$(dirname "${BASH_SOURCE[0]}")/.."
-		DOCKER_ARGS=("buildx" "build" "--platform=linux/amd64,linux/arm64")
-	elif [[ ! $PUBLISH_SUFFIX ]] && [[ ! $have_docker_container_driver ]]; then
-		DOCKER_ARGS=("buildx" "build" "--platform=linux/amd64")
-		>&2 echo "WARNING: heroku-24 and newer images are multi-arch images," \
-			"but this script is building single architecture images" \
-			"due to limitations of the current platform." \
-			"To build a multi-arch image, enable the 'containerd'" \
-			"snapshotter in Docker Desktop and/or use a 'docker-container'" \
-			"Docker BuildKit driver."
-	else
-		>&2 echo "ERROR: Can't build images with this configuration. Enable" \
-			"the 'containerd' snapshotter in Docker Desktop, enable" \
-			"the 'docker-container' driver in Docker, or use this script" \
-			"in build-only mode (don't provide PUBLISH_SUFFIX argument)."
-		abort 1
+	# heroku/heroku:24 and beyond support amd64 and arm64 and are published
+	# as multi-arch images.
+	if (( ${#ARCHS[@]} == 0 )); then
+		ARCHS=("amd64" "arm64")
 	fi
 	# heroku/heroku:24 and beyond images include CNB specific
 	# modifications, so separate *cnb* variants are not created.
 	VARIANTS=("-build:")
 fi
 
-if [[ $PUBLISH_SUFFIX ]]; then
-	# If there is a tag suffix, this script is pushing to a remote registry.
-	DOCKER_ARGS+=("--push")
-else
-	# Otherwise, load the image into the local image store.
-	DOCKER_ARGS+=("--load")
+if (( ${#ARCHS[@]} > 1 )) && (( STACK_VERSION <= 22 )) ; then
+	>&2 echo "ERROR: Can't build multi-arch images for heroku-22 and prior." \
+		"Provide a single target architecture or use a newer stack version."
+	abort 1
 fi
+if (( ${#ARCHS[@]} > 1 )) && [[ ! $have_containerd_snapshotter ]] ; then
+	>&2 echo "ERROR: Can't build images with this configuration. Enable" \
+		"the 'containerd' snapshotter in Docker Desktop, or specify a" \
+		"single target architecture"
+	abort 1
+fi
+
 
 write_package_list() {
 	local image_tag="$1"
 	local dockerfile_dir="$2"
-
-	# Extract the stack version from the dockerfile_dir variable (e.g., heroku-24)
-	local stack_version
-	stack_version=$(echo "$dockerfile_dir" | sed -n 's/^heroku-\([0-9]*\).*$/\1/p')
-
-	local archs=("amd64")
-	# heroku-24 and newer are multiarch. If containerd is available,
-	# the package list for each architecture can be generated.
-	if (( stack_version >= 24 )); then
-		if [[ $have_containerd_snapshotter ]]; then
-			archs+=(arm64)
-		else
-			>&2 echo "WARNING: Generating package list for single architecture." \
-				"Use the 'containerd' snapshotter to generate package lists" \
-				"for all architectures."
-		fi
-	fi
 	local output_file=""
-	for arch in "${archs[@]}"; do
-		if (( stack_version >= 24 )); then
-			output_file="${dockerfile_dir}/installed-packages-${arch}.txt"
-		else
+	for arch in "${ARCHS[@]}"; do
+		if (( STACK_VERSION <= 22 )) && [[ $arch = "amd64" ]]; then
 			output_file="${dockerfile_dir}/installed-packages.txt"
```

Comment:
(optional) I'd be fine with renaming `installed-packages.txt` to `installed-packages-amd64.txt` for Heroku-20/22 if you preferred to remove this special-casing. (If there are any third-party consumers of these files, they may actually prefer consistent naming across stacks too, I suppose.)

Reply:
This did occur to me a few times; I just didn't pull the trigger to keep the diff clean. Added in eed3aa2.